### DISCUSSION TITLE
Page config change to hard coded values is too restrictive

### DISCUSSION THREAD

**Discussion by shaneosullivan** [vercel/next.js#41234] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.2
      npm: 8.1.2
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.3.0
      eslint-config-next: 12.3.0
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome 105.0.5195.125

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

A recent change to NextJS broke the ability to have non-hard coded values in the Page config and points the developer to https://nextjs.org/docs/messages/invalid-page-config, e.g. 

```
const IS_MOBILE_SITE = process.env.SITE_TYPE === 'mobile'
export const config = {
  unstable_runtimeJS: !IS_MOBILE_SITE,
};
```

This is too restrictive, and forces horrible workarounds where the developer needs to have different page configurations depending on environment variables.
<img width="602" alt="Screenshot 2022-10-06 at 12 59 47" src="https://user-images.githubusercontent.com/49378/194307226-24ef23f4-1058-46c9-8001-a5f74647b051.png">

### Expected Behavior

Either revert to the previously accepted behavior of accepting variables for page config values, or at least allow setting the page config value to an environment variable value, which is available at build time.

### Link to reproduction

https://github.com/shaneosullivan/nextjs-invalid-page-config-issue

### To Reproduce

Check out the code at the linked repo and run `npm run dev`.  Open `localhost:3000` and in the terminal you will see the error described.  

On that web page, if the issue was not present, clicking the button would not show an alert box, but as the config is not recognized, clicking the button does show an alert.

**Comment by SukkaW** [comment]
This is the intended behavior (I am the one implementing this warning).

The exported `config` is statically analyzed during the build. It is will be stripped and will not be included in the final bundle, thus it will not be evaluated in the runtime.

**Comment by shaneosullivan** [comment]
I understand that this is the intended behavior since the recent changes to NextJS.  My point, not well made, is that in making this change you are not fully understanding the users needs and should try to find a way to address those needs, as outlined here.

Given the files are being statically analyzed at build time, you should be able to figure out how to do so using the values in environment variables, as those are available at build time also.

**Comment by shaneosullivan** [comment]
My specific need is that I have two sites, one for desktop and one for mobile.  On the desktop I ship the full JS, on mobile I do not ship JS.  Until a recent version of NextJS this was simply accomplished.  Is there a way to accomplish that with your recent changes?

**Comment by podviaznikov** [comment]
Plus one for this.

My use case is even more complex. I want 

```
export const config = {
  unstable_runtimeJS: false,
};
```

to be a function.

**Comment by shaneosullivan** [comment]
@SukkaW were these types of use cases considered before making the recent change to disallow them? If they were considered, why were they discarded as blockers?  If they were not considered, do you feel they mean that the change was in error and that another solution is needed to achieve whatever it is you are trying to improve by removing this capability?
